From 2536c31714700902830f2fb5bdb307ebd039e19c Mon Sep 17 00:00:00 2001
From: Tony Paul <Tony_Paul@comcast.com>
Date: Mon, 29 Nov 2021 15:06:24 +0000
Subject: [PATCH] Subject: [PATCH] DELIA-25802: 4K VOD AAMP PTS error after
 seek

qtdemux overrides segment.position value of a
received SEGMENT event. This was resulting in PTS error from brcmvideodecoder
since AAMP sends a GST_EVENT_SEGMENT right after flushing gst pipeline for seek.

Source: COMCAST
License: LGPL-2.1-or-later
Upstream-Status: Pending

Signed-off-by: Vinish K B <Vinish_KB@comcast.com>
---
 gst/isomp4/qtdemux.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gst/isomp4/qtdemux.c b/gst/isomp4/qtdemux.c
index e1a6ebf..5750031 100644
--- a/gst/isomp4/qtdemux.c
+++ b/gst/isomp4/qtdemux.c
@@ -2377,7 +2377,7 @@ gst_qtdemux_handle_sink_event (GstPad * sinkpad, GstObject * parent,
       if (!demux->upstream_format_is_time) {
         /* accept upstream's notion of segment and distribute along */
         segment.format = GST_FORMAT_TIME;
-        segment.position = segment.time = segment.start;
+        segment.time = segment.start;
         segment.duration = demux->segment.duration;
         segment.base = gst_segment_to_running_time (&demux->segment,
             GST_FORMAT_TIME, demux->segment.position);
-- 
2.14.2

